### PR TITLE
perf testing app for binary snaopshot for odsp driver

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

A small perf testing app for odsp binary/json format snapshot. Currently works for fluid documents in office.com and onenote.com. It does not work with sharing links. It fetches snapshot 5 times for a url and display the results.

Screenshot 2022-11-21 110745

### CODE REVIEW

File: examples/apps/snapshotFetchPerfApp/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      fallback:{
+          "url": require.resolve("url"),
+          "buffer": require.resolve("buffer/"), // note: the trailing slash is important!
+          "os": require.resolve("os-browserify/browser"),
+          "path": require.resolve("path-browserify"),
+          "http": require.resolve("stream-http"),
+          "zlib": require.resolve("browserify-zlib"),
+          "crypto": require.resolve("crypto-browserify"),
+          "stream": require.resolve("stream-browserify"),
+          "constants": require.resolve("constants-browserify"),
```
Comment (NicholasCouri) [inline examples/apps/snapshotFetchPerfApp/webpack.config.js:44]: do we need all of them ?  or any ?

File: examples/apps/snapshotFetchPerfApp/.eslintignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*.spec.ts
```
Comment (NicholasCouri) [inline examples/apps/snapshotFetchPerfApp/.eslintignore:1]: nit: need it ?

File: examples/apps/snapshotFetchPerfApp/src/fetchApp.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    div: HTMLDivElement,
+    odspAccessToken: string,
+) {
+    const binaryDiv = document.createElement("div");
+    binaryDiv.style.minHeight = "400px";
+    const binaryText = document.createElement("div");
+    binaryText.textContent = "Paste Your URL for Binary Snapshot Here";
+    binaryText.style.minHeight = "25px";
+    binaryText.style.marginBottom = "5px";
+    binaryText.style.marginTop = "10px";
```
Comment (NicholasCouri) [inline examples/apps/snapshotFetchPerfApp/src/fetchApp.ts:22]: nit: usually these settings live in a separate .css file and we would import them here. But as it is a test app I'd say it is not required

File: examples/apps/snapshotFetchPerfApp/src/fetchApp.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                mockLogger,
+                undefined,
+            );
+        }
+        fetchButtonClick(mockLogger, binaryDiv);
+    };
+
+    fetchButton2.onclick = async () => {
+        const resolvedUrl = await urlResolver.resolve({ url: text2.value });
+        assert(resolvedUrl !== undefined, 0x31a /* resolvedUrl is undefined */);
```
Comment (NicholasCouri) [inline examples/apps/snapshotFetchPerfApp/src/fetchApp.ts:75]: >0x31a  [](http://example.com/codeflow?start=42&length=6)

should we replace with a text ?

**Review by NicholasCouri — APPROVED** [review]
:shipit:

File: examples/apps/snapshotFetchPerfApp/webpack.config.js (hunk 1 above)
Comment (jatgarg) [reply examples/apps/snapshotFetchPerfApp/webpack.config.js]: no. removed.

**Pull request merged**